src/fabric.c

```diff
@@ -853,6 +957,14 @@ struct fi_info *DEFAULT_SYMVER_PRE(fi_dupinfo)(const struct fi_info *info)
 			if (dup->fabric_attr->prov_name == NULL)
 				goto fail;
 		}
+		if (info->handle && info->handle->fclass == FI_CLASS_NIC) {
+			int ret;
+			ret = ofi_dup_nic((struct fid_nic*) info->handle,
```

Comment:
You're not using the return value, and you don't need the cast. This can be expressed clearer as: 

```c
if (ofi_dup_nic(info->handle, &dup->handle) < 0)
    goto fail;
```